src/bitmessagemain.py

```diff
@@ -233,66 +259,82 @@ def start(self):
+        # close the main program even if there are threads left
+        singleWorkerThread.daemon = True
         singleWorkerThread.start()
 
         # Start the SQL thread
         sqlLookup = sqlThread()
-        sqlLookup.daemon = False  # DON'T close the main program even if there are threads left. The closeEvent should command this thread to exit gracefully.
+        # DON'T close the main program even if there are threads left.
+        # The closeEvent should command this thread to exit gracefully.
+        sqlLookup.daemon = False
         sqlLookup.start()
 
-        Inventory() # init
-        Dandelion() # init, needs to be early because other thread may access it early
+        Inventory()  # init
+        # init, needs to be early because other thread may access it early
+        Dandelion()
 
         # SMTP delivery thread
-        if daemon and BMConfigParser().safeGet("bitmessagesettings", "smtpdeliver", '') != '':
+        if daemon and BMConfigParser().safeGet(
+                "bitmessagesettings", "smtpdeliver", ''):
             smtpDeliveryThread = smtpDeliver()
             smtpDeliveryThread.start()
 
         # SMTP daemon thread
-        if daemon and BMConfigParser().safeGetBoolean("bitmessagesettings", "smtpd"):
+        if daemon and BMConfigParser().safeGetBoolean(
+                "bitmessagesettings", "smtpd"):
             smtpServerThread = smtpServer()
             smtpServerThread.start()
 
         # Start the thread that calculates POWs
         objectProcessorThread = objectProcessor()
-        objectProcessorThread.daemon = False  # DON'T close the main program even the thread remains. This thread checks the shutdown variable after processing each object.
+        # DON'T close the main program even the thread remains.
+        # This thread checks the shutdown variable after processing
+        # each object.
+        objectProcessorThread.daemon = False
         objectProcessorThread.start()
 
         # Start the cleanerThread
         singleCleanerThread = singleCleaner()
-        singleCleanerThread.daemon = True  # close the main program even if there are threads left
+        # close the main program even if there are threads left
+        singleCleanerThread.daemon = True
         singleCleanerThread.start()
 
         shared.reloadMyAddressHashes()
         shared.reloadBroadcastSendersForWhichImWatching()
 
-        if BMConfigParser().safeGetBoolean('bitmessagesettings', 'apienabled'):
-            try:
-                apiNotifyPath = BMConfigParser().get(
-                    'bitmessagesettings', 'apinotifypath')
-            except:
-                apiNotifyPath = ''
-            if apiNotifyPath != '':
-                with shared.printLock:
-                    print('Trying to call', apiNotifyPath)
+        singleAPIThread = singleAPI()
+        singleAPIThread.daemon = True
+        singleAPIThread.start()
```

Comment:
It should only run when the api is enabled.

Reply:
I moved that check into `singleAPI.__init__`

Reply:
For configuring the API yes, but not for starting the thread.

Reply:
It will not start if `apienaled = true` not set in config, look at line 200 and `_getApiAddress` (re)definition

Reply:
This is a little non-obvious. But it will still start the thread, even if it quits right away. Perhaps the init should raise an exception and then these 3 lines be wrapped in try/except instead.

Reply:
Thread will be stopped then because [serve_forever](/Bitmessage/PyBitmessage/blob/master/src/api.py#L61) not called

Reply:
What is the point of starting it when it quits right away? Isn't it better not to start it in the first place?